### DISCUSSION TITLE
Lambda functions on strings, ...

### DISCUSSION THREAD

**Discussion by ErikJansenIRefact** [duckdb/duckdb#9752] [Ideas]
As an example we need to apply a check on Dutch BSN numbers. A BSN should be of length 9 with only numeric characters.
The second test is that a should comply with a BSN-11 test. 

This is how we can now implement this in DuckDB (a column bsn exists):
list_aggregate([array_extract(bsn,x)::int*(if(x=9,-1,10-x)) for x in range(1,10,1)],'sum')%11 = 0

But if would have been perhaps more convenient to be able to iterate (as a reduce function to sum the product) over the BSN string character per character and apply a lambda function with the character and index.

**Comment by Tishj** [comment]
There's work being done on `list_reduce` I believe
